app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -431,8 +431,8 @@ class BrowserTabViewModel @Inject constructor(
             object FinishTrackerAnimation : DaxCommand()
             class HideDaxDialog(val cta: Cta) : DaxCommand()
         }
-        class InjectCredentials(val url: String, val credentials: LoginCredentials) : Command()
-        class CancelIncomingAutofillRequest(val url: String) : Command()
+        data class InjectCredentials(val url: String, val credentials: LoginCredentials) : Command()
```

Comment:
why the introduction of `data` here? (noting that none of the other commands are defined that way)

Reply:
Mainly to simplify the comparison. But I can maybe just change this to check the attributes instead of exact instance, just for the sake of consistency. Is there a reason why these are not data classes?

Reply:
Functionally, everything would work if they were all data classes but it would be a _slight_ unnecessary overhead to generate the extra things data classes generate when we mostly never need to use them again.

They are generally used as a one-time indication an event has happened (and with an associated transfer of data about that event), and not typically used for equality.